core/src/lib.rs

```diff
@@ -323,10 +329,28 @@ impl MainContext {
             "Retrieving case information for additional sequence cases...",
             &ctx,
         );
-        cases.extend(
-            Self::download_case_infos_no_sequence(additional, client, concurrent, self.pb())
-                .await?,
-        );
+        let additional_cases =
+            Self::download_case_infos_no_sequence(additional, client, concurrent, self.pb()).await;
+        for case in additional_cases {
+            match (case, &abort_behavior) {
+                (Ok(new_case), _) => {
+                    cases.insert(new_case);
+                }
+                (Err(e), args::SequenceErrorHandling::Abort) => {
+                    return Err(anyhow!(
+                        "Aborting due to error retrieving case in sequence: {e}"
+                    ));
+                }
+                (Err(e), args::SequenceErrorHandling::Continue) => {
+                    warn!("Could not retrieve case in sequence: {e}\nContinuing anyway.");
+                }
+                (Err(e), args::SequenceErrorHandling::Ask) => match self.ask_abort(e) {
+                    Ok(()) => abort_behavior = SequenceErrorHandling::Continue,
```

Comment:
For consistency with the Continue without asking case above, this should probably also produce the same log message